src/languages/index.js

```diff
@@ -50,24 +48,34 @@ function getTranslationByKey(translations, key, lang) {
     : `Invalid translation format for key '${key}' in language '${lang}'.`;
 }
 
-const interpolateText = (text, props = {}) => {
+const interpolateText = (text, params = {}) => {
   let result = text;
-
-  Object.entries(props).forEach(([key, value]) => {
+  Object.entries(params).forEach(([key, value]) => {
     result = result.replaceAll(`{{${key}}}`, value);
   });
-
   return result;
 };
 
-function translateLanguage(key, params = {}) {
-  const translations = loadTranslations(botLanguage);
-  if (!translations || !Object.values(translations).length) {
-    return `Translations not available for language '${botLanguage}'.`;
+function translateLanguage(key, params = {}, language = botLanguage) {
+  const translations = translationsCache[language] || {};
+  if (!Object.keys(translations).length) {
+    return `Translations not available for language '${language}'.`;
   }
-
-  const translation = getTranslationByKey(translations, key, botLanguage);
+  const translation = getTranslationByKey(translations, key, language);
   return interpolateText(translation, params);
 }
 
-module.exports = { translateLanguage };
+const translateCommand = (command) => {
+  const result = {};
+
+  for (const locale of locales) {
+    const translations = translationsCache[locale];
+    result[locale] = translations
+      ? getTranslationByKey(translations, command, locale)
+      : `Missing translation for ${locale}`;
+  }
```

Comment:
Here you can use the `translateLanguage` function, so you avoid repeating logic, following the principle DRY